File: src/main/python/Player.py
class Player():

    def requestStringFromPlayer(self,request):
        return input(request)

    def requestIntegerFromPlayer(self,request):
        valid = False
        while not valid:
            try:
                answerAsString = self.requestStringFromPlayer(request)
                answerAsInteger = int(answerAsString)
                valid = True
            except ValueError:
                print("Please enter an integer")
        return answerAsInteger

    def getPlayerPosition(self,colour,maxPosition):
        pointToMoveFrom = self.requestIntegerFromPlayer("Please enter position to move for " + colour)
        while not pointToMoveFrom <= maxPosition:
            print("Postion does not exist, please try again")
            pointToMoveFrom = self.requestIntegerFromPlayer("Please enter position to move for " + colour)
        return pointToMoveFrom

    def getPlayerDieToMove(self,colour,dice):
        numberOfMoves = self.requestIntegerFromPlayer("Please enter number of spaces to move for " + colour)
        # We want to check the player enters one of the die
        while numberOfMoves not in dice:
            print("Please select from",dice)
            numberOfMoves = self.requestIntegerFromPlayer("Please enter number of spaces to move for " + colour)
        return numberOfMoves

    def displayToPlayer(self,message="",ending=None):
        if ending == None:
            print(message)
        else:
            print(message,end=ending)

File: src/main/java/src/Player.java
package java.src;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.src.PlayerInterface;

public class Player implements PlayerInterface{
    private String requestStringFromPlayer(String request) {

        Scanner userInput = new Scanner(System.in);
        String answer = userInput.nextLine();
        return answer;
    }

    private Integer requestIntegerFromPlayer(String request) {
        int result = 0;
        Boolean valid = false;

        do {
            try {
                result = Integer.parseInt(requestStringFromPlayer(request));
                valid = true;
            } catch (NumberFormatException exception) {
                // Output expected NumberFormatException.
                System.out.println("Please enter a integer");
            }
        } while (!valid);

        return result;
    }

    private String formatList(ArrayList<Integer> aList){
        String newString = "";
        String preFix = "";
        for (int item: aList){
            if (newString != "") {
                preFix = ",";
            }
            newString += preFix + Integer.toString(item);
        }
        return newString;
    }

    public Integer getPlayerPosition(String colour,Integer maxPosition) {
        Integer pointToMoveFrom = requestIntegerFromPlayer("Please enter position to move for " + colour);
        while (pointToMoveFrom <= maxPosition) {
            System.out.println("Postion does not exist, please try again");
            pointToMoveFrom = requestIntegerFromPlayer("Please enter position to move for " + colour);
        }
        return pointToMoveFrom;
    }

    public Integer getPlayerDieToMove(String colour, ArrayList<Integer> dice) {
        int numberOfMoves = requestIntegerFromPlayer("Please enter number of spaces to move for " + colour);
        String diceLeft = formatList(dice);
        while ( !dice.contains(numberOfMoves)) {
            System.out.println("Please select from " + diceLeft);
            numberOfMoves = requestIntegerFromPlayer("Please enter number of spaces to move for " + colour);
        }
        return numberOfMoves;
    }

    public void displayToPlayer(String message, String ending) {
        if (message == null){
            message = " ";
        }
        if (ending == null) {
            System.out.println(message);
        }
        else{
            System.out.print(message + ending);
        }
    }

}

